tools/tpm2_pkcs11/utils.py

```diff
@@ -370,3 +371,45 @@ def check_pss_signature(tpm2, pctx, pauth):
             )
 
             return False
+
+class TSSPrivKey(univ.Sequence):
+    '''
+    Parse the ASN1 Sequence and provide the public and private blobs
+    ASN1_SEQUENCE(TSSPRIVKEY) = {
+    ASN1_SIMPLE(TSSPRIVKEY, type, ASN1_OBJECT),
+    ASN1_EXP_OPT(TSSPRIVKEY, emptyAuth, ASN1_BOOLEAN, 0),
+    ASN1_SIMPLE(TSSPRIVKEY, parent, ASN1_INTEGER),
+    ASN1_SIMPLE(TSSPRIVKEY, pubkey, ASN1_OCTET_STRING),
+    ASN1_SIMPLE(TSSPRIVKEY, privkey, ASN1_OCTET_STRING)
+    } ASN1_SEQUENCE_END(TSSPRIVKEY)
+    '''
+ 
+    componentType = namedtype.NamedTypes(
+        namedtype.NamedType('type', univ.ObjectIdentifier()),
+        namedtype.OptionalNamedType('emptyauth', univ.Boolean().subtype(
+               explicitTag=tag.Tag(tag.tagClassContext, tag.tagFormatSimple, 0)
+           )),
+        namedtype.NamedType('parent', univ.Integer()),
+        namedtype.NamedType('pubkey', univ.OctetString()),
+        namedtype.NamedType('privkey', univ.OctetString())
+    )
+
+def asn1parse_tss_key(keypath):
```

Comment:
@AndreasFuchsSIT, @pdxjohnny and @tstruk, I wanted to show you this magic if you need to parse a TSS ASN1 key format in Python into it's components.

Reply:
Whoa, nice! Thanks for sharing! Reminds me of https://www.python.org/dev/peps/pep-0593/#combining-runtime-and-static-uses-of-annotations I saw the other day which looks to be coming in Python 3.9.